### PR TITLE
[streamalert] Remove samples, update schemas and add first CloudTrail rule

### PR DESCRIPTION
**Pull request by mime-frame**
to: @jacknagz 
cc: @airbnb/streamalert-maintainers 

**What**
- Add new and updated log schemas
- Remove sample logs and rules
- Add first CloudTrail rule
- Remove tabs from `conf/sources.json`

**Why**
- Address bugs in the old CarbonBlack schema
- Allow folks to write CloudTrail rules
- In it's current form, samples make it harder for new users to deploy streamalert
- Pave the way for committing more production rules (value out of the box)

**Testing**

`./test/scripts/rule_test.sh`

### CODE REVIEW

File: conf/logs.json [hunk 1]
```diff
@@ -182,100 +463,200 @@
...
+        "unknown4": "string",
+        "actionflags": "string",
+        "srcloc": "string",
+        "dstloc": "string",
+        "pkts_sent": "string",
+        "pkts_received": "string"
       }
-    }
+    },
+    "hints": {
```
Comment (jacknagz) [inline conf/logs.json]: This should be `"configuration": {"log_patterns": {"type": ["TRAFFIC"]}}`

File: conf/sources.json [hunk 2]
```diff
@@ -1,38 +1,20 @@
...
-				"json_log",
-				"syslog_log",
-				"kv_log",
-				"csv_log",
-				"osquery",
-				"cloudtrail"
-			]
-		}
-	},
-	"sns": {
```
Comment (jacknagz) [inline conf/sources.json]: why remove the example SNS topic?

File: rules/cloudtrail/cloudtrail_critical_api.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
+from stream_alert.rule_processor.rules_engine import StreamRules
+from rules.helpers.base import in_set
+
+rule = StreamRules.rule
+disable = StreamRules.disable()
+
+
+@rule(logs=['cloudtrail:api_events'],
+      matchers=[],
+      outputs=['aws-s3:sample.bucket'])
```
Comment (jacknagz) [inline rules/cloudtrail/cloudtrail_critical_api.py]: it should be `aws-s3:sample_bucket`

File: stream_alert/rule_processor/main.py [hunk 4]
```diff
@@ -18,9 +18,6 @@
 import os
 
 from stream_alert.rule_processor.handler import StreamAlert
-from rules import (
```
Comment (jacknagz) [inline stream_alert/rule_processor/main.py]: if we are getting rid of this, we need an action item to standardize loading matchers from within the loop below

File: conf/logs.json (hunk 1 above)
Comment (mime-frame) [inline conf/logs.json]: I guess internal is broken as well, noted

File: conf/sources.json (hunk 2 above)
Comment (mime-frame) [inline conf/sources.json]: goal is to remove all samples

File: rules/cloudtrail/cloudtrail_critical_api.py (hunk 3 above)
Comment (mime-frame) [inline rules/cloudtrail/cloudtrail_critical_api.py]: i guess existing rules were broken too?

Comment (ryandeivert) [inline rules/cloudtrail/cloudtrail_critical_api.py]: I have a bunch of these I'm fixing now in other rules - please adhere to what jack is suggesting with `aws-s3:sample_bucket`

File: conf/sources.json (hunk 2 above)
Comment (jacknagz) [inline conf/sources.json]: then why retain the s3 input?

File: conf/logs.json (hunk 1 above)
Comment (jacknagz) [inline conf/logs.json]: it's unused

Comment (mime-frame) [inline conf/logs.json]: but if used, broken, noted

File: conf/sources.json (hunk 2 above)
Comment (mime-frame) [inline conf/sources.json]: its required for rule integration testing?

File: stream_alert/rule_processor/main.py (hunk 4 above)
Comment (mime-frame) [inline stream_alert/rule_processor/main.py]: i'll just re-add it back, better suited for another PR

**Comment by mime-frame** [comment]
@jacknagz suggestions addressed except for your s3 comment, see my reply. Thanks! :)

**Pull request merged**